src/passphrase/validation.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import Mnemonic from 'bip39';
+
+export const countPassphraseWhitespaces = passphrase => {
+	const whitespaceMatches = passphrase.match(/\s/g);
+	return whitespaceMatches ? whitespaceMatches.length : 0;
+};
+
+export const countPassphraseWords = passphrase =>
+	passphrase.split(' ').filter(Boolean).length;
+
+export const countUppercaseCharacters = passphrase => {
+	const uppercaseCharacterMatches = passphrase.match(/[A-Z]/g);
+	return uppercaseCharacterMatches ? uppercaseCharacterMatches.length : 0;
```

Comment:
I'm not aware of your guidelines, but this function could be shorter
```javascript
// option 1
const uppercaseCharacterMatches = passphrase.match(/[A-Z]/g) || [];
return uppercaseCharacterMatches.length;
// option 2, just in one line. But not sure about readability.
export const countUppercaseCharacters = passphrase => (passphrase.match(/[A-Z]/g) || []).length;
```

The same could be applied to the function `countPassphraseWhitespaces `

Reply:
Hi @Reshetnyak, thanks for the input, I actually almost suggested option 2 in my review (I always like a direct return from an arrow function 🙂 ). In this case I think readability/intelligibility is clearest with the longer option currently in place.